### PR TITLE
add cross compiler linux-mingw-w64

### PR DESCRIPTION
**Pull request by NeroBurner**
add cross compiler mingw-w64 from linux to windows

related to Issue #153

### CODE REVIEW

File: linux-mingw-w64.cmake [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "Unix Makefiles"
+)
+
+include("${CMAKE_CURRENT_LIST_DIR}/utilities/polly_common.cmake")
+
+# need to set system name for cross compiling from linux to windows
+set(CMAKE_SYSTEM_NAME Windows)
+set(CROSS_COMPILE_TOOLCHAIN_PREFIX "x86_64-w64-mingw32")
+
+set(TOOLCHAIN_PREFIX x86_64-w64-mingw32)
```
Comment (ruslo) [inline linux-mingw-w64.cmake]: Do we need this variable?

File: linux-mingw-w64.cmake [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+include(
+    "${CMAKE_CURRENT_LIST_DIR}/compiler/gcc-cross-compile-simple-layout.cmake"
+)
+include("${CMAKE_CURRENT_LIST_DIR}/flags/cxx11.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/flags/static.cmake")
+
+# set fortran compiler (part of mingw-w64)
+set(CMAKE_Fortran_COMPILER "${TOOLCHAIN_PATH_AND_PREFIX}-gfortran" CACHE PATH "gfortran" )
```
Comment (ruslo) [inline linux-mingw-w64.cmake]: I think it's okay to move it to the `compiler/gcc-cross-compile.cmake`

**Comment by NeroBurner** [comment]
updated, please review

**Comment by NeroBurner** [comment]
where do I add the additional ci dependencies `sudo apt-get install mingw-w64 g++-mingw-w64`?

**Comment by ruslo** [comment]
> where do I add the additional ci dependencies sudo apt-get install mingw-w64 g++-mingw-w64?

First add just this info to documentation: Toolchains -> linux-mingw-w64. Then I want to try this toolchain locally, if it's ok will test `.travis.yml`.

**Comment by ruslo** [comment]
> sudo apt-get install mingw-w64 g++-mingw-w64

On my machine package `g++-mingw-w64` installed by `mingw-w64`. So it can be just `sudo apt-get install mingw-w64`.

**Comment by ruslo** [comment]
We need `linux-mingw-w64` entry in `toolchain_table.py`, Linux section:
* https://github.com/ruslo/polly/blob/46dba7175b9f63af04fcc4cc238dc1bd1d0be88c/bin/detail/toolchain_table.py#L249-L254

**Comment by ruslo** [comment]
> Then I want to try this toolchain locally, if it's ok will test .travis.yml

Looks good:
* https://github.com/travis-ci-tester/travis-test-mingw/blob/d8eafc0b31b733020bcb5b35c9256c41ef6b2755/.travis.yml#L16-L19
* https://travis-ci.org/travis-ci-tester/travis-test-mingw/builds/245668694#L535

Since `addons: apt: packages: - mingw-w64` will affect all linux toolchain in polly/.travis.yml my suggestion is to leave it as is for now. This will be covered by pkg.boost in Hunter.

**Comment by NeroBurner** [comment]
updated:
- add documentation for linux-mingw-w64
- add linux-mingw-w64 to toolchain_table

**Pull request merged**